test-data/run.test

```diff
@@ -870,3 +870,41 @@ assert increment_attr(P()).x == 1
 assert increment_attr_o(O()).x == 2
 assert increment_all_indices([1, 2, 3]) == [2, 3, 4]
 assert increment_all_keys({'a':1, 'b':2, 'c':3}) == {'a':2, 'b':3, 'c':4}
+
+[case testNestedFunctions]
+from typing import Callable
+
+def outer() -> Callable[[], object]:
+    def inner() -> object:
+        return None
+    return inner
+
+def first() -> Callable[[], Callable[[], str]]:
+    def second() -> Callable[[], str]:
+        def third() -> str:
+            return 'third: nested function'
+        return third
+    return second
+
```

Comment:
Add some tests that pass arguments to the functions.

Reply:
Done.